Mathlib/NumberTheory/LSeries/AbstractFuncEq.lean

```diff
@@ -0,0 +1,471 @@
+
+lemma hf_modif_FE (x : ℝ) (hx : 0 < x) :
+    P.f_modif (1 / x) = (P.ε * ↑(x ^ P.k)) • P.g_modif x := by
+  rcases lt_trichotomy 1 x with hx' | rfl | hx'
+  · have : 1 / x < 1 := by rwa [one_div_lt hx one_pos, div_one]
+    rw [f_modif, Pi.add_apply, indicator_of_not_mem (not_mem_Ioi.mpr this.le),
+      zero_add, indicator_of_mem (mem_Ioo.mpr ⟨div_pos one_pos hx, this⟩), g_modif, Pi.add_apply,
+      indicator_of_mem (mem_Ioi.mpr hx'), indicator_of_not_mem
+      (not_mem_Ioo_of_ge hx'.le), add_zero, P.h_feq _ hx, smul_sub]
+    simp_rw [rpow_neg (one_div_pos.mpr hx).le, one_div, inv_rpow hx.le, inv_inv]
+  · rw [div_one, f_modif, Pi.add_apply, indicator_of_not_mem, indicator_of_not_mem,
+      g_modif, Pi.add_apply, indicator_of_not_mem, indicator_of_not_mem]
+    simp only [add_zero, smul_zero]
+    all_goals { simp }
+  · have : 1 < 1 / x := by rwa [lt_one_div one_pos hx, div_one]
+    rw [f_modif, Pi.add_apply, indicator_of_mem (mem_Ioi.mpr this),
+      indicator_of_not_mem (not_mem_Ioo_of_ge this.le), add_zero, g_modif, Pi.add_apply,
+      indicator_of_not_mem (not_mem_Ioi.mpr hx'.le),
+      indicator_of_mem (mem_Ioo.mpr ⟨hx, hx'⟩), zero_add, P.h_feq _ hx, smul_sub]
+    simp_rw [rpow_neg hx.le, ← mul_smul]
+    field_simp [(rpow_pos_of_pos hx P.k).ne', P.hε]
+
+/-- Given a weak FE-pair `(f, g)`, modify it into a strong FE-pair by subtracting suitable
+correction terms from `f` and `g`. -/
+def modifStrongFEPair : StrongFEPair E where
+  hf_int   := P.hf_modif_int
+  hg_int   := P.symm.hf_modif_int
+  h_feq    := P.hf_modif_FE
+  hε       := P.hε
+  hk       := P.hk
+  hf₀      := rfl
+  hg₀      := rfl
+  hf_top r := by
+    refine (P.hf_top r).congr' ?_ (by rfl)
+    filter_upwards [eventually_gt_atTop 1] with x hx
+    rw [f_modif, Pi.add_apply, indicator_of_mem (mem_Ioi.mpr hx),
+      indicator_of_not_mem (not_mem_Ioo_of_ge hx.le), add_zero, sub_zero]
+  hg_top r := by
+    refine (P.hg_top r).congr' ?_ (by rfl)
+    filter_upwards [eventually_gt_atTop 1] with x hx
+    rw [f_modif, Pi.add_apply, indicator_of_mem (mem_Ioi.mpr hx),
+      indicator_of_not_mem (not_mem_Ioo_of_ge hx.le), add_zero, sub_zero]
+    rfl
+
+/- Alternative form for the difference between `f - f₀` and its modified term. -/
+lemma f_modif_aux1 : EqOn (fun x ↦ P.f_modif x - P.f x + P.f₀)
+    ((Ioo 0 1).indicator (fun x : ℝ ↦ P.f₀ - (P.ε * ↑(x ^ (-P.k))) • P.g₀)
+    + ({1} : Set ℝ).indicator (fun _ ↦ P.f₀ - P.f 1)) (Ioi 0) := by
+  intro x (hx : 0 < x)
+  simp_rw [f_modif, Pi.add_apply]
+  rcases lt_trichotomy x 1 with hx' | rfl | hx'
+  · simp_rw [indicator_of_not_mem (not_mem_Ioi.mpr hx'.le), zero_add,
+      indicator_of_mem (mem_Ioo.mpr ⟨hx, hx'⟩)]
+    rw [indicator_of_not_mem (mem_singleton_iff.not.mpr hx'.ne), add_zero]
+    abel
+  · simp_rw [indicator_of_not_mem (not_mem_Ioi.mpr le_rfl), zero_add,
+      indicator_of_not_mem (not_mem_Ioo_of_ge le_rfl), zero_sub, zero_add]
+    rw [indicator_of_mem (mem_singleton _)]
```

Comment:
```suggestion
  · simp [add_comm, sub_eq_add_neg]
```
would do it (or `simp; abel`, but this would run afoul of the non-terminal `simp` linter, I guess...).
Not sure it's really better, though.